### PR TITLE
Fix react-emotion typing

### PR DESCRIPTION
**Pull request by Ailrun**
**What**: Fixing react-emotion typing

**Why**: Current typing is redundant.

**How**: Using `create-emotion-styled` typing

**Checklist**:
- [N/A] Documentation
- [x] Tests
- [x] Code complete

### CODE REVIEW

**Comment by Ailrun** [comment]
Why could not yarn fetch packages?

**Comment by mitchellhamilton** [comment]
The yarn registry is down right now. https://status.yarnpkg.com/

File: packages/react-emotion/types/index.d.ts [hunk 1]
```diff
@@ -1,120 +1,21 @@
...
+  Themed,
+} from 'create-emotion-styled';
+
+export {
+  CreateStyled,
+  Interpolation,
+  StyledComponent,
+  StyledOptions,
+  Themed,
+} from 'create-emotion-styled';
```
Comment (mitchellhamilton) [inline packages/react-emotion/types/index.d.ts]: Was importing everything twice intentional?

Comment (Ailrun) [reply packages/react-emotion/types/index.d.ts]: @mitchellhamilton Oh... no, it's my mistake. Thank you for pointing this!

**Comment by mitchellhamilton** [comment]
Is there anything that should be updated in the ts docs? https://github.com/emotion-js/emotion/blob/master/docs/typescript.md

**Comment by Ailrun** [comment]
@mitchellhamilton Hm... I will check it.
However, there's other problem I missed. It looks like `styled` could take other StyledComponent as well, am I right?

**Comment by mitchellhamilton** [comment]
Yes, styled can take styled components.

**Comment by Ailrun** [comment]
@mitchellhamilton I didn't know that... Does it works with `emotion` as well?
I mean,
```tsx
import { css } from 'emotion';
import styled from 'react-emotion';
const Comp = styled.div({});

const x = css`
  ${Comp} {
    color: blue;
  }
`;
```

is also valid?

**Comment by mitchellhamilton** [comment]
Yes, that also works.

**Comment by Ailrun** [comment]
@mitchellhamilton Then I should update other typings first and revise this PR. I close this now.

**Comment by LukasWerfel** [comment]
[This](https://emotion.sh/docs/styled#targeting-another-emotion-component) still doesnt seem to work with TS.
```
Type error: A computed property name must be of type 'string', 'number', 'symbol', or 'any'. 
```

**Pull request closed without merging**